### ISSUE TITLE
How to update pre-trained(Saved) model in Keras

### ISSUE THREAD

**Issue by PonraJS-21** [keras-team/keras#14234]
I would like to know how to update an existing model with, new dataset without train with old dataset. Is there any way to Update model with only new dataset

Here I try to update the model which was trained with only 2 class and third class empty, then update 3rd class too...

#!/usr/bin/env python
# coding: utf-8

# In[74]:

import numpy as np
import matplotlib.pyplot as plt
get_ipython().run_line_magic('matplotlib', 'inline')

# keras imports for the dataset and building our neural network
from keras.datasets import mnist
from keras.models import Sequential
from keras.layers import Conv2D, Flatten, Dense, Dropout, MaxPooling2D
from keras.utils import to_categorical
from keras.models import load_model

# In[75]:

(X_train, y_train), (X_test, y_test) = mnist.load_data()

# In[122]:

def create_dataset(mnist_img_num, tot_img = 10):
    x_arr = []
    y_arr = []
    for i in range(len(X_test)):
        if(y_train[i] == mnist_img_num[0] or y_train[i] == mnist_img_num[1]):
            x_arr.append(X_train[i])
            y_arr.append(y_train[i])
            tot_img-=1
        if(tot_img == 0):
            break
    return np.asarray(x_arr), np.asarray(y_arr)

# ## Initially Training a model with 1 and 2 handwritten digits

# In[123]:

x, y = create_dataset([1 , 2], 500)

# In[124]:

y

# In[125]:

y = to_categorical(y)

# In[126]:

y[:5]

# In[127]:

for i in range(1):
    plt.imshow(x[i])
    plt.show()

# In[128]:

model = Sequential()
model.add(Conv2D(32, (3, 3), activation='relu', input_shape=(28, 28, 1)))
model.add(MaxPooling2D((2, 2)))
model.add(Conv2D(64, (3, 3), activation='relu'))
model.add(MaxPooling2D((2, 2)))
model.add(Conv2D(64, (3, 3), activation='relu'))
model.add(Flatten())
model.add(Dense(64, activation='relu'))
model.add(Dense(3, activation='softmax'))

# In[129]:

model.compile(optimizer='rmsprop', loss='categorical_crossentropy', metrics=['accuracy'])

# In[130]:

x = x.reshape((len(y), 28, 28, 1))

# In[131]:

x.shape

# In[135]:

for _ in range(10):
    history = model.train_on_batch(x,y)

# In[136]:

history

# In[137]:

model.save('partially_trained.h5')

# ## Prediction for 2

# In[138]:

test = np.expand_dims(x[5], axis=0)
print('Prediction: ',np.argmax(model.predict(test)[0]))
print('Actual: ', np.argmax(y[5]))
print(y[5])

# ## Prediction for 1

# In[139]:

test = np.expand_dims(x[0], axis=0)
print('Prediction: ',np.argmax(model.predict(test)[0]))
print('Actual: ',np.argmax(y[0]))
print(y[0])

# In[140]:

model = load_model('partially_trained.h5')

# # Re-Training with 0 and 2 digits

# In[141]:

x, y = create_dataset([0 , 2], 500)
y = to_categorical(y)
x = x.reshape((len(y), 28, 28, 1))
print(x.shape)

# In[142]:

# loaded_model.compile(optimizer='rmsprop', loss='categorical_crossentropy', metrics=['accuracy'])

# In[144]:

for _ in range(10):
    history_2 = model.train_on_batch(x,y)

# In[145]:

history_2

# ## Prediction for 0

# In[146]:

testx, testy = create_dataset([1 , 0], 5)
print(testy)
testx = testx.reshape((len(testy), 28, 28, 1))
print(testx.shape)
test = np.expand_dims(testx[0], axis=0)
print(test.shape)
print('predicted: ', np.argmax(loaded_model.predict(test)))
print('actual: ',testy[0])

# In[147]:

plt.imshow(test[0].reshape((28,28)))
# plt.suptitle(y[i])
plt.show()

# ## Prediction for 1

# In[148]:

testx, testy = create_dataset([1 , 2], 5)
print(testy)
testx = testx.reshape((len(testy), 28, 28, 1))
print(testx.shape)
test = np.expand_dims(testx[0], axis=0)
print(test.shape)
print('predicted: ', np.argmax(loaded_model.predict(test)))
print('actual: ',testy[0])

# In[149]:

plt.imshow(test[0].reshape((28,28)))
# plt.suptitle(y[i])
plt.show()

# ## Prediction for 2

# In[150]:

testx, testy = create_dataset([1 , 2], 5)
print(testy)
testx = testx.reshape((len(testy), 28, 28, 1))
print(testx.shape)
test = np.expand_dims(testx[1], axis=0)
print(test.shape)
print('predicted: ', np.argmax(loaded_model.predict(test)))
print('actual: ',testy[1])

# In[151]:

plt.imshow(test[0].reshape((28,28)))
plt.show()

# In[ ]:

**Comment by PonraJS-21** [comment]
### Output

[0 1 1 1 1]
(5, 28, 28, 1)
(1, 28, 28, 1)
predicted:  2
actual:  0
